apps/astarte_realm_management/lib/astarte_realm_management/engine.ex

```diff
@@ -937,6 +941,25 @@ defmodule Astarte.RealmManagement.Engine do
     end
   end
 
+  defp verify_mappings_max_storage_retention(realm_name, interface) do
+    with {:ok, max_retention} when is_integer(max_retention) <-
```

Comment:
This will fallthrough with `{:ok, value}` if `max_retention` is not an integer. Is this what you want? I don't think so since the caller expects `:ok / {:error, reason}`